src/peer_connection.cpp

```diff
@@ -4954,6 +4962,13 @@ namespace libtorrent {
 		time_duration const time_limit = seconds(
 			m_settings.get_int(settings_pack::inactivity_timeout));
 
+		// if we are close enough to the limit, consider the peer connection
+		// list full. This will enable the inactive timeout
+		bool const max_session_conns = m_ses.num_connections()
+			>= m_settings.get_int(settings_pack::connections_limit) - 5;
+		bool const max_torrent_conns = t && t->num_peers()
+			>= t->max_connections() - 5;
```

Comment:
this is the second part of the logic change, to treat the connection slots at limit slightly before reaching the absolute limit